### DISCUSSION TITLE
How to add username or other fields in JWT sub data?

### DISCUSSION THREAD

**Discussion by TianaNanta** [fastapi-users/fastapi-users#1319] [Q&A] [answered]
![Screenshot from 2023-11-11 09-18-02](https://github.com/fastapi-users/fastapi-users/assets/75852322/31d33265-f819-4bde-a550-ede55fe80434)
What is the problem with this JWT Token that says Invalid signature?
And how can I add other field in the sub data but not only the UUID? I want username and user_type to be also stored in the Token
Thanks in advance

**Comment by hgalytoby** [comment, marked as answer]
I think we can overlook the 'Invalid signature' issue displayed on your JWT page.

Are you using the JWTStrategy?

If you are using JWTStrategy, you can do the following:

```python
class MyJWTStrategy(JWTStrategy):
    async def write_token(self, user: models.UP) -> str:
        data = {"sub": str(user.id), "aud": self.token_audience, 'user_type': 'user_type'}
        return generate_jwt(
            data, self.encode_key, self.lifetime_seconds, algorithm=self.algorithm
        )


def get_jwt_strategy() -> MyJWTStrategy:
    return MyJWTStrategy(secret=SECRET, lifetime_seconds=3600)


auth_backend = AuthenticationBackend(
    name="jwt",
    transport=bearer_transport,
    get_strategy=get_jwt_strategy,
)
```

For the rest of the code, please refer to [this link](https://fastapi-users.github.io/fastapi-users/12.1/configuration/full-example/#sqlalchemy).

If I misunderstood, please let me know.

**Reply by TianaNanta** [reply]
Thank you, it works
